### PR TITLE
ddl, executor: fix rename table compatibility

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
The results of `alter table old_name rename new_name` and `rename table old_name to new_name` are different.

```
mysql> show variables like "%lower_case_table_names%";
+------------------------+-------+
| Variable_name          | Value |
+------------------------+-------+
| lower_case_table_names | 2     |
+------------------------+-------+
1 row in set (0.01 sec)
mysql> show tables;
+--------------+
| Tables_in_db |
+--------------+
| t            |
| t1           |
+--------------+
2 rows in set (0.00 sec)

mysql>  rename table t to T;
ERROR 1050 (42S01): Table 'T' already exists
mysql>  alter table t rename T;
Query OK, 0 rows affected (0.00 sec)

mysql> show tables;
+--------------+
| Tables_in_db |
+--------------+
| t            |
| t1           |
+--------------+
2 rows in set (0.00 sec)

```

### What is changed and how it works?
We treat `alter table old_name rename new_name` and `rename table old_name to new_name` differently.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8709)

### CODE REVIEW

File: executor/ddl.go [hunk 1]
```diff
@@ -122,7 +122,8 @@ func (e *DDLExec) executeRenameTable(s *ast.RenameTableStmt) error {
 	}
 	oldIdent := ast.Ident{Schema: s.OldTable.Schema, Name: s.OldTable.Name}
 	newIdent := ast.Ident{Schema: s.NewTable.Schema, Name: s.NewTable.Name}
-	err := domain.GetDomain(e.ctx).DDL().RenameTable(e.ctx, oldIdent, newIdent)
+	isAlterTable := false
+	err := domain.GetDomain(e.ctx).DDL().RenameTable(e.ctx, oldIdent, newIdent, isAlterTable)
```
Comment (jackysp) [inline executor/ddl.go]: Why not pass false directly?

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1184,7 +1184,8 @@ func (d *ddl) AlterTable(ctx sessionctx.Context, ident ast.Ident, specs []*ast.A
 			err = d.AlterColumn(ctx, ident, spec)
 		case ast.AlterTableRenameTable:
 			newIdent := ast.Ident{Schema: spec.NewTable.Schema, Name: spec.NewTable.Name}
-			err = d.RenameTable(ctx, ident, newIdent)
+			isAlterTable := true
+			err = d.RenameTable(ctx, ident, newIdent, isAlterTable)
```
Comment (jackysp) [inline ddl/ddl_api.go]: ditto

File: executor/ddl.go (hunk 1 above)
Comment (winkyao) [reply executor/ddl.go]: Agree.

Comment (zimulala) [reply executor/ddl.go]: This makes it more readable.

**Comment by zimulala** [comment]
PTAL @jackysp @winkyao

**Comment by zimulala** [comment]
PTAL @jackysp @winkyao

File: executor/ddl.go (hunk 1 above)
Comment (winkyao) [reply executor/ddl.go]: I don't think it is necessary.

**Pull request merged**